erpnext/public/js/controllers/accounts.js

```diff
@@ -246,6 +247,30 @@ if(!erpnext.taxes.flags[cur_frm.cscript.tax_table]) {
 		cur_frm.cscript.validate_taxes_and_charges(cdt, cdn);
 	});
 
+	frappe.ui.form.on(cur_frm.cscript.tax_table, "category", function(frm, cdt, cdn) {
+		var d = locals[cdt][cdn];
+		var open_form = frm.open_grid_row();
+		if(open_form) {
+			erpnext.taxes.update_category_and_add_deduct_tax(open_form);
+		} else {
+			// apply in current row
+			erpnext.taxes.update_category_and_add_deduct_tax(frm.get_field('taxes').grid.get_row(cdn));
+		}
+		refresh_field('add_deduct_tax', d.name, 'taxes');
```

Comment:
You already have the `open_form` object, no need for the `d` object, you can just do:

```suggestion
		refresh_field('add_deduct_tax', open_form.doc.name, 'taxes');
```